osbs/utils/yaml.py

```diff
@@ -72,7 +78,17 @@ def read_yaml(yaml_data, schema, package=None):
             if path.startswith('.'):
                 path = path[1:]
 
-            logger.error('validation error (%s): %s', path or 'at top level', error.message)
-        raise
+            # get all context messages without duplicates caused by 'anyOf' validator
+            error_contexts = set()
+            for context in error.context:
+                error_contexts.add(context.message)
+
+            error_message = "{}: validating '{}' has failed ({})".format(
+                            path or 'at top level', error.validator,
+                            ", ".join(error_contexts) or error.message)
+
+            logger.error(error_message)
+
+        raise OsbsValidationException(exc_message)
```

Comment:
You misunderstood me. I wanted to debug log exception, but this new exception should contain error message in the same format as from line 86 (and only for error in exception, i.e. first issue)